### PR TITLE
Feat/migrate sui react initial props to typescript

### PR DESCRIPTION
**Pull request by midudev**
Migrate @s-ui/react-initial-props to TypeScript.

- [x] We should adapt @s-ui/lint to be able to work with TypeScript files or create a @s-ui/ts-lint.

### CODE REVIEW

**Review by kikoruiz — APPROVED** [review]

**Review by arnau-rius — APPROVED** [review]
👏🏽 👏🏽 👏🏽

File: packages/sui-react-initial-props/package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "main": "lib/index.js",
```
Comment (andresz1) [inline packages/sui-react-initial-props/package.json:4]: Should [typings](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package) field be added here ? For example:
```
{
  "typings": "dist/index.d.ts",
}
```

File: packages/sui-react-initial-props/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
```
Comment (andresz1) [inline packages/sui-react-initial-props/tsconfig.json:2]: Should this config be shared somehow in the feature across our packages ?

**Review by andresz1 — APPROVED** [review]

**Review by rmoralp — APPROVED** [review]

**Review by jcger — APPROVED** [review]

**Review by desko27 — APPROVED** [review]
This is awesome

**Review by jcger — APPROVED** [review]

**Review by kikoruiz — APPROVED** [review]

**Review by rmoralp — APPROVED** [review]

**Pull request merged**